packages/e2e/test/stake-pool/StakePoolProvider.test.ts

```diff
@@ -0,0 +1,220 @@
+        );
+    });
+
+    describe('identifier filter', () => {
+      it('using id as value', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using name as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.name));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without name: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ name: pool.metadata!.name }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using ticker as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.ticker));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without ticker: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ ticker: pool.metadata!.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id OR ticker as value (same pool)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }, { ticker: pools[0].metadata?.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id OR ticker as value (distinct pools)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }, { ticker: pools[1].metadata?.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(2);
+      });
+
+      it('using id AND ticker as value (same pool)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: {
+            identifier: { _condition: 'and', values: [{ id: pools[0].id }, { ticker: pools[0].metadata?.ticker }] }
+          },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id AND ticker as value (distinct pools)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: {
+            identifier: { _condition: 'and', values: [{ id: pools[0].id }, { ticker: pools[1].metadata?.ticker }] }
+          },
+          pagination
+        });
+        expect(totalResultCount).toBe(0);
+      });
+    });
+
+    describe('pledgeMet filter', () => {
```

Comment:
These tests aren't very portable. Could maybe test that response `totalResultCount` of 
`pledgeMet=true` + `pledgeMet=false` = `pledgeMet=undefined`

Reply:
test added